packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts

```diff
@@ -213,9 +213,11 @@ export class InsuredBridgeL1Client {
     return this.bridgePools[l1Token];
   }
 
-  getBridgePoolForL2Token(l2Token: string): BridgePoolData {
-    const bridgePoolData = Object.values(this.bridgePools).filter((bridgePool) => bridgePool.l2Token == l2Token)[0];
-    if (!bridgePoolData) throw new Error(`No bridge pool initialized for ${l2Token}`);
+  getBridgePoolForL2Token(l2Token: string, chainId: string): BridgePoolData {
+    const bridgePoolData = Object.values(this.bridgePools).find((bridgePool) => {
```

Comment:
I think you might need to remove the braces around the inner function or add a "return". Not 100% sure, but I think the current syntax will cause the inner function to always return undefined.